### PR TITLE
add terminal initialization form

### PR DESCRIPTION
**Pull request by christianvogt**
**Fixes**: 
https://issues.redhat.com/browse/ODC-3579

**Analysis / Root cause**: 
Identifying automatically the correct namespace to create a DevWorkspace resource in for a user for the cloud shell is difficult and has several concerns to take into consideration such as security and resource quotas. For an interim solution, we prompt the user to choose their namespace as a workaround.

**Solution Description**: 
* Find DevWorkspace CR using a label selector to see if the user has a previously created workspace.
* If no previous workspace, display a form to the user:
  * Select or create a namespace.
    * Ideally only present the user with namespaces they have correct access to and can create a DevWorkspace in.
  * On submit, create the new workspace in the chosen namespace.
* If the user lacks permissions display a message as such.
* Display the cloud shell terminal.

**Screen shots / Gifs for design review**: 
(gif is out of date with text casing and nav header button showing active state)
cli

**Tests**
image

***To Test***
Clone repo `https://github.com/che-incubator/che-workspace-operator`
Run the following commands:
```
export WEBHOOK_ENABLED=true
export DEFAULT_ROUTING=openshift-oauth
make deploy
```
(current as per writing this description but instructions are taken directly from the README.md in the repo)

**Browser conformance**: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [x] Edge

### CODE REVIEW

**Comment by christianvogt** [comment]
cc @serenamarie125 @parvathyvr @abhinandan13jan

File: frontend/packages/console-app/src/components/cloud-shell/setup/CloudShellSetupForm.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  FormikProps<FormikValues>,
+  'errors' | 'handleSubmit' | 'handleReset' | 'status' | 'isSubmitting'
+>> = ({ errors, handleSubmit, handleReset, status, isSubmitting }) => {
+  return (
+    <Form onSubmit={handleSubmit} className="co-m-pane__form">
+      <NamespaceSection />
+      <FormFooter
+        handleReset={handleReset}
+        errorMessage={status && status.submitError}
+        isSubmitting={isSubmitting}
```
Comment (rohitkrai03) [inline frontend/packages/console-app/src/components/cloud-shell/setup/CloudShellSetupForm.tsx:18]:
```suggestion
        isSubmitting={status?.isSubmitting}
```

File: frontend/packages/console-app/src/components/cloud-shell/setup/CloudShellSetupForm.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        submitLabel="Start"
+        disableSubmit={!_.isEmpty(errors) || isSubmitting}
```
Comment (rohitkrai03) [inline frontend/packages/console-app/src/components/cloud-shell/setup/CloudShellSetupForm.tsx:20]:
```suggestion
        disableSubmit={!_.isEmpty(errors) || status?.isSubmitting}
```

File: frontend/packages/console-app/src/components/cloud-shell/setup/CloudShellSetup.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        });
+      }
+      await k8sCreate(
+        WorkspaceModel,
+        newCloudShellWorkSpace(createCloudShellResourceName(), namespace, username),
+      );
+      onSubmit && onSubmit();
+    } catch (err) {
+      actions.setSubmitting(false);
+      actions.setStatus({ submitError: err.message });
```
Comment (rohitkrai03) [inline frontend/packages/console-app/src/components/cloud-shell/setup/CloudShellSetup.tsx:57]: Probably a good idea to also set `isSubmitting: false` in status. And then you won't need `actions.setSubmitting(false)`

File: frontend/packages/console-app/src/components/cloud-shell/setup/CloudShellSetup.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+          metadata: {
+            name: namespace,
+          },
+        });
+      }
+      await k8sCreate(
+        WorkspaceModel,
+        newCloudShellWorkSpace(createCloudShellResourceName(), namespace, username),
+      );
+      onSubmit && onSubmit();
```
Comment (rohitkrai03) [inline frontend/packages/console-app/src/components/cloud-shell/setup/CloudShellSetup.tsx:54]: Need to add `actions.setStatus({ isSubmitting: false });` if the requests were successful.

**Comment by christianvogt** [comment]
All comments from @rohitkrai03 are regarding `isSubmitting`. He had issues with the `actions.isSubmitting` and therefore set a custom status. The reason was because the loading indicator wasn't showing up. I tried my changes using `actions.isSubmitting` and the indicator is showing up as expected. So i do nothing I need to change to use `status.isSubmitting`.

File: frontend/packages/console-app/src/components/cloud-shell/setup/CloudShellSetup.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  activeNamespace,
+  onSubmit,
+  onCancel,
+  username,
+}) => {
+  const initialValues: CloudShellSetupFormData = {
+    namespace: activeNamespace === ALL_NAMESPACES_KEY ? undefined : activeNamespace,
+  };
+  const handleSubmit = async (values: CloudShellSetupFormData, actions) => {
+    actions.setStatus({ isSubmitting: true });
```
Comment (rohitkrai03) [inline frontend/packages/console-app/src/components/cloud-shell/setup/CloudShellSetup.tsx:38]: You don't need to set status like this if you're using actions.isSubmitting.

Comment (rohitkrai03) [inline frontend/packages/console-app/src/components/cloud-shell/setup/CloudShellSetup.tsx:38]:
```suggestion
    actions.isSubmitting(true);
```

Comment (christianvogt) [reply frontend/packages/console-app/src/components/cloud-shell/setup/CloudShellSetup.tsx]: Interestingly enough it still showed up. I guess formik automatically sets submitting. Regardless I've updated the code.

**Pull request merged**